tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -545,139 +545,105 @@ class TestLinearOperator(unittest.TestCase):
     # modified from scipy
     # class that defines parametrized custom cases
     # adapted from scipy's analogous tests
-    def _inner_cases(self, original, xp, sp, dtype):
-        if(self.inner_modification == 'normal'):
-            return (xp.array(original, dtype=dtype), original)
-        if(self.inner_modification == 'sparse'):
-            return (sp.csr_matrix(
-                    original, dtype=dtype), original)
-
+    def _inner_cases(self, xp, sp, A):
         # creating base-matrix-like class with default
         # matrix-vector and adjoint-matrix-vector impl
-        def mv(x, dtype):
-            y = original.dot(x)
-            if len(x.shape) == 2:
-                y = y.reshape(-1, 1)
-            return y
-
-        def rmv(x, dtype):
-            return original.T.conj().dot(x)
 
-        # defining the base-class
-        class BaseMatlike(sp.linalg.LinearOperator):
-            args = ()
+        def mv(x):
+            return A.dot(x)
 
-            def __init__(self, dtype):
-                self.dtype = xp.dtype(dtype)
-                self.shape = original.shape
+        def rmv(x):
+            return A.T.conj().dot(x)
 
-            def _matvec(self, x):
-                return mv(x, self.dtype)
-
-        # using above base-class to make and test sub-classes based
-        # on the current `inner_modification` parameter
-        if(self.inner_modification == 'adjoint'):
+        # defining the user-defined classes
+        class BaseMatlike(sp.linalg.LinearOperator):
 
-            # defining the sub-class
-            class HasAdjoint(BaseMatlike):
-                args = ()
+            def __init__(self):
+                self.dtype = A.dtype
+                self.shape = A.shape
 
-                def _adjoint(self):
-                    shape = self.shape[1], self.shape[0]
-                    matvec = partial(rmv, dtype=self.dtype)
-                    rmatvec = partial(mv, dtype=self.dtype)
-                    return sp.linalg.LinearOperator(matvec=matvec,
-                                                    rmatvec=rmatvec,
-                                                    dtype=self.dtype,
-                                                    shape=shape)
-            return (HasAdjoint(dtype), original)
+            def _adjoint(self):
+                shape = self.shape[1], self.shape[0]
+                return sp.linalg.LinearOperator(
+                    matvec=rmv, rmatvec=mv, dtype=self.dtype, shape=shape)
 
-        # defining the sub-classes
-        class HasRmatvec(BaseMatlike):
```

Comment:
Removed the test for this class because scipy supports `_rmatvec` only for backward compatibility currently.
https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.LinearOperator.html